### PR TITLE
fix: Update EUI version and switch to Amsterdam

### LINKED ISSUE #167: [Bug] Use full Amsterdam theme
### Bug summary

Recently, the EUI squad helpfully pointed out that the recorder's using the older EUI theme rather than the latest Amsterdam one. We should work with them to get the recorder running the latest versions of the theme.

The [have documented](https://elastic.github.io/eui/#/guidelines/getting-started) a getting started experience, which we can use as a baseline for making sure our usage of the framework is in line with their expectations.

### Recorder Version

latest

### Steps to reproduce

Run the recorder, see outdated UI.

### Expected behavior

Run the recorder, see Amsterdam UI.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by justinkambic**
## Summary

Resolves #167.

As noted in the linked issue, we aren't using the latest and greatest EUI theme. Fixing that here.

## Implementation details

There may be minor things that break and are tough to notice. It might be worthwhile to build a binary from `main` and compare it against this branch in dev mode.

## How to validate this change

There should be no functional change with this patch, it should update EUI and utilize the Amsterdam styles.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -24,14 +24,10 @@ THE SOFTWARE.
...
 import React, { useContext } from "react";
 import { useEffect, useState } from "react";
-import {
-  EuiCode,
-  EuiEmptyPrompt,
-  EuiThemeProvider,
-  EuiThemeAmsterdam,
-} from "@elastic/eui";
+import { EuiCode, EuiEmptyPrompt, EuiProvider } from "@elastic/eui";
+import createCache from "@emotion/cache";
```
Comment (vigneshshanmugam) [inline src/App.tsx:28]: I don't see the @emotion/cache on the deps list.

File: src/App.tsx [hunk 2]
```diff
@@ -54,6 +49,13 @@ import { ExportScriptFlyout } from "./components/ExportScriptFlyout";
 import { useRecordingContext } from "./hooks/useRecordingContext";
 import { StartOverWarningModal } from "./components/StartOverWarningModal";
 
+const cache = createCache({
```
Comment (vigneshshanmugam) [inline src/App.tsx:52]: can you add some comments on how its used and whats its for?

File: src/App.tsx (hunk 1 above)
Comment (justinkambic) [reply src/App.tsx:28]: We already had it as a dep of `@emotion/react`, which I guess is why it worked. I've added it as a direct dep. d8e107d7f2e77f14e8ae730e55a06162befeb814

File: src/App.tsx (hunk 2 above)
Comment (justinkambic) [reply src/App.tsx:57]: Basically there are issues with EUI that necessitate this. I've added a comment and a link to the relevant documentation. 0d71c584cfc307996d086f1e261fdaa818368723

Comment (vigneshshanmugam) [reply src/App.tsx:57]: Thanks they helps

**Review by vigneshshanmugam — APPROVED** [review]
LGTM

**Pull request merged**